### ISSUE TITLE
IndexError: tuple assignment index out of range

### ISSUE THREAD

**Issue by ddi98** [dashingsoft/pyarmor#1096]
after obfuscating my flask project i get the following error:

```
(venv) PS C:\Users\Giuseppe\Dropbox\PROGETTI_PYCHARM\FormulariTools\dist\formularitools> python app.py
Traceback (most recent call last):
  File "<frozen __main__>", line 3, in <module>
  File "<frozen FormulariTools.app>", line 40, in <module>
  File "<frozen routes_vivifir>", line 3, in <module>
IndexError: tuple assignment index out of range
```

**Comment by jondy** [comment]
I'll appreciated if you could provide a simple script to reproduce this issue.

**Comment by ddi98** [comment]
I've located the part of the function that generates the problem but I can't fix it. if I remove the fields under request.method == 'GET' it works
`    if form.validate_on_submit():
        pass


    elif request.method == 'GET':
        form.autorizza_invio_pec_prod.process_data(istanza_formulario.autorizza_invio_pec_prod)
        form.indirizzo_pec_produttore_sl.data = pec_produttore_sl
        form.indirizzo_pec_produttore_ul.data = pec_produttore_ul

        form.autorizza_invio_pec_inte.process_data(istanza_formulario.autorizza_invio_pec_inte)
        form.indirizzo_pec_intermediario_sl.data = pec_intermediario_sl
        form.indirizzo_pec_intermediario_ul.data = pec_intermediario_ul

        if dati_formulario['PROD_FORMU']:
            form.cod_produttore.data = dati_formulario['PROD_FORMU']
            form.des_produttore.data = dati_formulario['PROD_SL_DES']
            form.piva_produttore.data = dati_formulario['PROD_SL_PIVA']
            form.cf_produttore.data = dati_formulario['PROD_SL_CODFIS']
            form.city_produttore.data = dati_formulario['PROD_UL_COM']
            form.ind_produttore.data = dati_formulario['PROD_UL_IND'] + f" {dati_formulario['PROD_UL_NUMCIVIC']}"

        if dati_formulario['DATI_FORMU'] and (tabella == 'MOVIMENTI'):
            form.data_inizio_trasporto.data = datetime.strptime(dati_formulario['DATI_FORMU'], '%d-%m-%Y')
        else:
            form.data_inizio_trasporto.data = istanza_plico.parametri_griglia['DATA_INIZIO_TRASPORTO']

        if dati_formulario['ORAI_FORMU']:
            form.ora_inizio_trasporto.data = datetime.strptime(dati_formulario['ORAI_FORMU'], '%H:%M')

        if dati_formulario['DATF_FORMU'] and (tabella == 'MOVIMENTI'):
            form.data_fine_trasporto.data = datetime.strptime(dati_formulario['DATF_FORMU'], '%d-%m-%Y')
        else:
            form.data_fine_trasporto.data = istanza_plico.parametri_griglia['DATA_FINE_TRASPORTO']

        if dati_formulario['ORAF_FORMU'] and (tabella == 'MOVIMENTI'):
            form.ora_fine_trasporto.data = datetime.strptime(dati_formulario['ORAF_FORMU'], '%H:%M')
        else:
            form.ora_fine_trasporto.data = istanza_plico.parametri_griglia['ORA_FINE_TRASPORTO']

        if dati_formulario['COD_AUTIST'] and (tabella == 'MOVIMENTI'):
            form.cod_autista.data = dati_formulario['COD_AUTIST']
            form.des_autista.data = dati_formulario['DES_AUTIST']
        else:
            form.cod_autista.data = istanza_plico.parametri_griglia['COD_AUTISTA']
            form.des_autista.data = istanza_plico.parametri_griglia['DES_AUTISTA']

        if dati_formulario['COD_AUTOMEZ'] and (tabella == 'MOVIMENTI'):
            form.cod_automezzo.data = dati_formulario['COD_AUTOMEZ']
            form.targa_automezzo.data = dati_formulario['TARGA_AUTOMEZ']
        else:
            form.cod_automezzo.data = istanza_plico.parametri_griglia['COD_AUTOMEZZO']
            form.targa_automezzo.data = istanza_plico.parametri_griglia['TARGA_AUTOMEZZO']

        if dati_formulario['COD_RIMORCH'] and (tabella == 'MOVIMENTI'):
            form.cod_rimorchio.data = istanza_plico.parametri_griglia['COD_RIMORCH']
            form.targa_rimorchio.data = istanza_plico.parametri_griglia['TARGA_RIMORCH']
        else:
            form.cod_rimorchio.data = istanza_plico.parametri_griglia['COD_RIMORCHIO']
            form.targa_rimorchio.data = istanza_plico.parametri_griglia['TARGA_RIMORCHIO']


        form.kg_netto.data = dati_formulario['NETTO_FORMU'].replace(',','.')`

**Comment by KohnoseLami** [comment]
```python
Traceback (most recent call last):
  File "<frozen __main__>", line 3, in <module>
IndexError: tuple assignment index out of range
```

I bought a Pro, but I got a similar error and it doesn't encrypt properly.
Command used:
```
pyarmor gen test.py
```

Also, I erased some lines to try to determine the cause, but when I deleted one line or some random place, it worked fine.
It doesn't seem to be a bug caused by any particular place, (Bugs caused by various combinations, for example)

I can't post a code on Github, so I can't provide a code, but I want Pyarmor to work, so I'd like to provide a code if I have a way to contact you personally.

**Comment by jondy** [comment]
@ddi98 thanks, I'll check it.

@KohnoseLami could you email this test script to pyarmor@163.com, thanks.

**Comment by ddi98** [comment]
@jondy if needed, I can send you the whole project via email

**Comment by KohnoseLami** [comment]
> @ddi98 thanks, I'll check it.
> 
> @KohnoseLami could you email this test script to [pyarmor@163.com](mailto:pyarmor@163.com), thanks.

Sent, by kohnoselami@gmail.com.

**Comment by jondy** [comment]
@ddi98 thanks.  It's enough to send the problem script to pyarmor@163.com

**Comment by ddi98** [comment]
@jondy I sent you the whole project via dropbox link

**Comment by vzubin** [comment]
We encountered the following error while running our application after obfuscation
application is using falcon and gunicorn

```
Traceback (most recent call last):
  File "//anaconda3/envs/falcon_framework/lib/python3.10/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "//anaconda3/envs/falcon_framework/lib/python3.10/site-packages/gunicorn/workers/base.py", line 129, in init_process
    self.load_wsgi()
  File "/home//anaconda3/envs/falcon_framework/lib/python3.10/site-packages/gunicorn/workers/base.py", line 138, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/home//anaconda3/envs/falcon_framework/lib/python3.10/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/home//anaconda3/envs/falcon_framework/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 52, in load
    return self.load_wsgiapp()
  File "/home//anaconda3/envs/falcon_framework/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 41, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/home/l161dtrv/anaconda3/envs/falcon_framework/lib/python3.10/site-packages/gunicorn/util.py", line 350, in import_app
    __import__(module)
  File "<frozen Application.falcon_app>", line 3, in <module>
  File "<frozen Application.falcon_app>", line 43, in <module>
  File "<frozen Application.startup>", line 47, in autoload
  File "<frozen healthcheck_api>", line 3, in <module>
IndexError: tuple assignment index out of range
```

Can you please provide an estimated time for the fix?

**Comment by KohnoseLami** [comment]
> Reproduced it and I'm fixing it. As a quick workaround, try to use `--no-wrap` to obfuscate scripts

I tried that option and it worked.
What does this option refer to?
And it doesn't seem to work in combination with some options like mix-str and rtf.
```python
Traceback (most recent call last):
  File "<frozen __main__>", line 3, in <module>
  File "<frozen main>", line 1, in <module>
NameError: name '__assert_armored__' is not defined
```

**Comment by jondy** [comment]
`--no-wrap` is simple mode to obfuscate the script, it's a temporary solution. 

I have found the problem and will fix it in next release. Once it's fixed, `--no-wrap` is not necessary.

Please open new issue to report new bug, if this bug has been reported by others and it is still open, comment it.

**Comment by jondy** [comment]
Fixed in `8.1.dev7`
